src/ring_theory/polynomial/chebyshev/dickson.lean

```diff
@@ -0,0 +1,127 @@
+/-
+Copyright (c) 2021 Julian Kuelshammer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Kuelshammer
+-/
+
+import data.polynomial.eval
+import ring_theory.polynomial.chebyshev.defs
+
+/-!
+# Dickson polynomials
+
+The Dickson polynomials are two families of polynomials indexed by `ℕ`,
+with coefficients in a commutative ring `R` depending on an element `a∈R`.
+They are closely related to the Chebyshev polynomials in the case that `a=1`.
+When `a=0` they are just the family of monomials `x ^ n`.
+
+## Main declarations
+
+* `polynomial.dickson₁`: the Dickson polynomials of the first kind.
+* `polynomial.dickson₂`: the Dickson polynomials of the second kind.
+
+## References
+
+* [R. Lidl, G. L. Mullen and G. Turnwald, _Dickson polynomials_][MR1237403]
+
+## TODO
+
+* Move the definition of `lambdashev` from `chebyshev.defs` into this file
+  (or even remove it because it is a special case of the polynomials defined here?).
+* Show that `lambdashev` is equal to the characteristic polynomial of the adjacency matrix of a
+  type A Dynkin diagram.
+-/
+
+noncomputable theory
+
+namespace polynomial
+
+variables (R S : Type*) [comm_ring R] [comm_ring S] (a : R)
+
+/-- `dickson₁` is the `n`-th Dickson polynomial of the first kind associated to the element `a∈R`.
+-/
+noncomputable def dickson₁ : ℕ → polynomial R
+| 0       := 2
+| 1       := X
+| (n + 2) := X * dickson₁ (n + 1) - (monomial 0 a) * dickson₁ n
+
+@[simp] lemma dickson₁_zero : dickson₁ R a 0 = 2 := rfl
+@[simp] lemma dickson₁_one : dickson₁ R a 1 = X := rfl
```

Comment:
May as well make `R` implicit, it can be inferred from `a`:
```suggestion
variables {R S : Type*} [comm_ring R] [comm_ring S] (a : R)

/-- `dickson₁` is the `n`-th Dickson polynomial of the first kind associated to the element `a∈R`.
-/
noncomputable def dickson₁ : ℕ → polynomial R
| 0       := 2
| 1       := X
| (n + 2) := X * dickson₁ (n + 1) - (monomial 0 a) * dickson₁ n

@[simp] lemma dickson₁_zero : dickson₁ a 0 = 2 := rfl
@[simp] lemma dickson₁_one : dickson₁ a 1 = X := rfl
```
etc

Reply:
Thanks for the suggestion. That makes sense.